micro_http/src/common/headers.rs

```diff
@@ -13,63 +12,226 @@ pub enum Header {
     ContentLength,
     /// Header `Content-Type`.
     ContentType,
+    /// Header `Expect`.
+    Expect,
+    /// Header `Transfer-Encoding`.
+    TransferEncoding,
 }
 
 impl Header {
-    fn raw(&self) -> &'static [u8] {
+    pub fn raw(&self) -> &'static [u8] {
         match self {
             Header::ContentLength => b"Content-Length",
             Header::ContentType => b"Content-Type",
+            Header::Expect => b"Expect",
+            Header::TransferEncoding => b"Transfer-Encoding",
+        }
+    }
+
+    fn try_from(string: &[u8]) -> Result<Self, RequestError> {
+        if let Ok(s) = String::from_utf8(string.to_vec()) {
```

Comment:
I would rename this variable to `utf8_string` or something else. `s` is not very suggestive.